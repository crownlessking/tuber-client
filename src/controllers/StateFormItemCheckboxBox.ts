import AbstractState from './AbstractState';
import IFormChoices from '../interfaces/IFormChoices';
import type StateFormItemCheckboxCustom from './templates/StateFormItemCheckboxCustom';
import StateFormItemCustom from './StateFormItemCustom';
import { CSSProperties } from 'react';
import { FormControlLabelProps } from '@mui/material';
import React from 'react';

export interface IStateFormItemCheckboxBox extends IFormChoices { }

export default class StateFormItemCheckboxBox
  extends AbstractState implements IStateFormItemCheckboxBox
{
  private _checkboxState: IStateFormItemCheckboxBox;
  private _parentDef: StateFormItemCheckboxCustom;
  private _checkboxHas?: StateFormItemCustom<this>;

  constructor(checkboxState: IStateFormItemCheckboxBox, parent: StateFormItemCheckboxCustom) {
    super()
    this._checkboxState = checkboxState;
    this._parentDef = parent;
  }

  get state(): IStateFormItemCheckboxBox { return this._checkboxState; }
  get parent(): StateFormItemCheckboxCustom { return this._parentDef; }
  get name(): string { return this._checkboxState.name ?? ''; }
  get label(): string { return this._checkboxState.label ?? ''; }
  get color(): Required<IStateFormItemCheckboxBox>['color'] {
    return this._checkboxState.color || 'default';
  }
  get disabled(): boolean|undefined { return this._checkboxState.disabled; }
  get props(): Record<string, unknown> { return this._checkboxState.props ?? {}; }
  get theme(): CSSProperties { return this.die('Not implemented yet.', {}); }
  get has(): StateFormItemCustom<this> {
    return this._checkboxHas || (
      this._checkboxHas = new StateFormItemCustom(
        this._checkboxState.has || {},
        this
      )
    );
  }
  get hasLabel(): boolean { return !!this._checkboxState.label; }
  get formControlLabelProps(): FormControlLabelProps {
    return this.has.formControlLabelProps ?? {
      'control': React.createElement('input', { type: 'checkbox' }),
      'label': this.label
    };
  }
}
