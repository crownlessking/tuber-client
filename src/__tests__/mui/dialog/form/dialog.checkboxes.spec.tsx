import renderer from 'react-test-renderer';
import DialogCheckboxes from '../../../../mui/dialog/form/dialog.checkboxes';
import StateFormItem from '../../../../controllers/StateFormItem';
import type StateForm from '../../../../controllers/StateForm';
import StateFormItemCheckboxBox from '../../../../controllers/StateFormItemCheckboxBox';

describe('src/mui/dialog/form/dialog.checkboxes.tsx', () => {

  const hive = {} as Record<string, unknown>;

  it('should render correctly', () => {
    const checkboxes = new StateFormItem<StateForm, StateFormItemCheckboxBox>({
      type: 'checkboxes',

      // [TODO] Finish implementing this to test.

    }, {} as StateForm);
    const tree = renderer
      .create(<DialogCheckboxes def={checkboxes} hive={hive} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});