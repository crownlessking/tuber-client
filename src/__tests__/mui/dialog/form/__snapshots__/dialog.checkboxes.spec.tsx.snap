// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/mui/dialog/form/dialog.checkboxes.tsx should render correctly 1`] = `null`;
